cmd/mirror-main.go

```diff
@@ -1041,6 +1068,10 @@ func runMirror(ctx context.Context, srcURL, dstURL string, cli *cli.Context, enc
 	mirrorSrcBuckets := srcClt.GetURL().Type == objectStorage && srcClt.GetURL().Path == string(srcClt.GetURL().Separator)
 	mirrorBucketsToBuckets := mirrorSrcBuckets && createDstBuckets
 
+	if cli.Bool("with-versioning") && (!checkIfBucketIsVersioned(ctx, srcURL) || !checkIfBucketIsVersioned(ctx, dstURL)) {
+		fatalIf(errInvalidArgument().Trace(cli.Command.Name), "You cannot specify --with-versioning in versioning not enabled buckets")
```

Comment:
A clearer and more grammatically correct version of the error message might be:
```suggestion
		fatalIf(errInvalidArgument().Trace(cli.Command.Name), "You cannot specify --with-versioning for buckets where versioning is not enabled.")
```